### PR TITLE
Features/add crate products

### LINKED ISSUE #49: Create crateProducts joins table
- Form the many-to-many relationship between `crates` and `products` by forming the `crateProducts` joins table.
- write appropriate queries (and test in GraphQL GUI)
- [x] crateProducts
- [x] crateProduct
- [x] crateProductsByCrate
- [x] crate ProductsByProduct

- write appropriate mutations (and test in GraphQL GUI) 
- [x] crateProductsCreate (requires admin auth)
- [x] crateProductsRemove (requires admin auth)

### PR DESCRIPTION
**Pull request by Strompy**
# Crate Pull Request Form

## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Created new DB join table: crateProducts
Table joins crates and products so that they will be associated for future crate subscriptions
Added GraphQL functions:
* Queries: crateProducts, crateProduct, crateProuductByCrate, crateProductByProduct
* Mutations: crateProductCreate, crateProductRemove

closes #49 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other, Please explain:
`Your message here`
## Has This Been Tested?
- [ ] No
- [X ] Yes
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
Tested in the GraphiQL gui, no automated testing yet

# Message for reviewer:
To add new tables run from the terminal
`$ npm db:migrate`

No seeds have been created at this time.
Create and Remove functions require admin authentication

# Checklist:
- [X] My code has no unused/commented out code
- [X] My code has no console logs
- [X] I have performed a self-review of my own code
- [X] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: code/api/src/modules/crateProduct/resolvers.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// App Imports
+import params from '../../config/params'
```
Comment (sienna-kopf) [inline code/api/src/modules/crateProduct/resolvers.js:2]: Thanks error handling on GraphQL GUI for telling us to include params here!

File: code/api/src/modules/crateProduct/resolvers.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    include: [
+      { model: models.Product, as: 'product' },
+      { model: models.Crate, as: 'crate' },
+    ]
+  })
+}
+
+// Create crateProduct
+export async function create(parentValue, { crateId, productId }, { auth }) {
+  if(auth.user && auth.user.role === params.user.roles.admin) {
```
Comment (sienna-kopf) [inline code/api/src/modules/crateProduct/resolvers.js:56]: So glad we could steal this admin authentication from the crate/product create!

File: code/api/src/modules/crateProduct/model.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+'use strict'
```
Comment (sienna-kopf) [inline code/api/src/modules/crateProduct/model.js:1]: I feel like this was surprisingly easy!!

**Review by sienna-kopf — APPROVED** [review]

**Review by rwilliams659 — APPROVED** [review]
Great work!

**Review by Abdeboskey — APPROVED** [review]
Good work, y'all have been such busy bees!

**Review by JoshSevy — APPROVED** [review]
Great work everyone. So cool to see how much momentum we have been gaining.

**Pull request merged**